tests/foreman/cli/test_realm.py

```diff
@@ -32,7 +32,10 @@
 
 
 @pytest.mark.tier1
-def test_positive_delete_by_name(module_fake_proxy):
+@pytest.mark.destructive
```

Comment:
With these tests being marked destructive, the `subscribe_satellite` fixture needs to be using `destructive_sat` instead of `default_sat`. 

This mark controls test collection, but doesn't control replacing `default_sat` instance.

The core patterns for this need some updates, it came up in the reviewers call this morning. @JacobCallahan @omkarkhatavkar

Reply:
Hm, so what to do with that? This applies to `enroll_idm_and_configure_external_auth`, `configure_realm` and `module_fake_proxy` as well, right?

Reply:
This does become complicated and redundant if you've got some destructive tests and some non-destructive tests using the same fixtures.  Hence why the pattern needs some updating.